File: account-manage/account-manage/src/main/java/com/service/account_manage/entity/id/AccountCodeDetailId.java
package com.service.account_manage.entity.id;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.*;

import java.io.Serializable;
import java.util.Objects;

@Embeddable
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AccountCodeDetailId implements Serializable {

    private String userId;

    private String codeId;

    private String codeGroup;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AccountCodeDetailId that = (AccountCodeDetailId) o;
        return Objects.equals(userId, that.userId) &&
                Objects.equals(codeId, that.codeId) &&
                Objects.equals(codeGroup, that.codeGroup);
    }

    @Override
    public int hashCode() {
        return Objects.hash(userId, codeId, codeGroup);
    }
}

File: app-server-account-manage/account_manage/src/main/kotlin/com/service/account_manage/entity/id/AccountCodeDetailId.kt
package com.service.account_manage.entity.id

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.io.Serializable

@Embeddable
data class AccountCodeDetailId(
    @Column(name = "USER_ID", length = 20, nullable = false)
    val userId : String?= null,

    @Column(name = "CODE_ID", length = 20, nullable = false)
    val codeId : String?= null,

    @Column(name = "CODE_GROUP", length = 20, nullable = false)
    val codeGroup : String?= null

): Serializable
